controllers/tls_workflow.go

```diff
@@ -133,24 +126,124 @@ func LinkGatewayToClusterIssuerFunc(objs controller.Store) machinery.LinkFunc {
+}
+
+func GetTLSPoliciesByEvents(topology *machinery.Topology, events []controller.ResourceEvent) []machinery.Policy {
+	policies := lo.Filter(topology.Policies().Items(), func(item machinery.Policy, index int) bool {
+		_, ok := item.(*kuadrantv1alpha1.TLSPolicy)
+		return ok
+	})
+
+	var affectedPolicies []machinery.Policy
+	for _, event := range events {
+		if event.Kind == machinery.GatewayGroupKind {
+			ob := event.NewObject
+			if ob == nil {
+				ob = event.OldObject
 			}
 
-			// Can infer linked gateways through the policy
-			linkedGateways := lo.Filter(gateways, func(g *gwapiv1.Gateway, index int) bool {
-				for _, l := range linkedPolicies {
-					if string(l.Spec.TargetRef.Name) == g.GetName() && g.GetNamespace() == l.GetNamespace() {
+			g := machinery.Gateway{Gateway: ob.(*gwapiv1.Gateway)}
+
+			affectedPolicies = append(affectedPolicies, lo.Filter(policies, func(item machinery.Policy, index int) bool {
+				for _, tg := range item.GetTargetRefs() {
+					if g.GetLocator() == tg.GetLocator() {
 						return true
 					}
 				}
-
 				return false
-			})
+			})...)
+		}
 
-			return lo.Map(linkedGateways, func(item *gwapiv1.Gateway, index int) machinery.Object {
-				return &machinery.Gateway{Gateway: item}
-			})
-		},
+		if event.Kind == kuadrantv1alpha1.TLSPolicyGroupKind {
+			ob := event.NewObject
+			if ob == nil {
+				ob = event.OldObject
+			}
+
+			affectedPolicies = append(affectedPolicies, lo.Filter(policies, func(item machinery.Policy, index int) bool {
+				return item.GetName() == ob.GetName() && item.GetNamespace() == ob.GetNamespace()
+			})...)
+		}
+
+		if event.Kind == CertManagerCertificateKind {
+			ob := event.NewObject
+			if ob == nil {
+				ob = event.OldObject
+			}
+
+			affectedPolicies = append(affectedPolicies, lo.Filter(policies, func(item machinery.Policy, index int) bool {
+				p := item.(*kuadrantv1alpha1.TLSPolicy)
+				return utils.IsOwnedBy(ob, p)
+			})...)
+		}
+
+		if event.Kind == CertManagerIssuerKind {
+			ob := event.NewObject
+			if ob == nil {
+				ob = event.OldObject
+			}
+
+			affectedPolicies = append(affectedPolicies, lo.Filter(policies, func(item machinery.Policy, index int) bool {
+				p := item.(*kuadrantv1alpha1.TLSPolicy)
+
+				return ob.GetName() == p.Spec.IssuerRef.Name && lo.Contains([]string{"", certmanagerv1.IssuerKind}, p.Spec.IssuerRef.Kind) &&
+					item.GetNamespace() == ob.GetNamespace()
+			})...)
+		}
+
+		if event.Kind == CertManagerClusterIssuerKind {
+			ob := event.NewObject
+			if ob == nil {
+				ob = event.OldObject
+			}
+
+			affectedPolicies = append(affectedPolicies, lo.Filter(policies, func(item machinery.Policy, index int) bool {
+				p := item.(*kuadrantv1alpha1.TLSPolicy)
+				return ob.GetName() == p.Spec.IssuerRef.Name && p.Spec.IssuerRef.Kind == certmanagerv1.ClusterIssuerKind
+			})...)
+		}
 	}
+
+	// Return only unique policies as there can be duplicates from multiple events
+	return lo.Uniq(affectedPolicies)
```

Comment:
Hmm. Maybe specify a uniqueness key and thus avoid possible problems with pointer comparison:

```suggestion
	return lo.UniqBy(affectedPolicies, func(p machinery.Policy) string { return p.GetLocator() } )
```